cl/recap/mergers.py

```diff
@@ -1508,13 +1511,38 @@ def merge_attachment_page_data(
                 )
             except ValueError:
                 pass
-        rd.save()
+        await rd.asave()
 
         # Do *not* do this async — that can cause race conditions.
-        add_items_to_solr([rd.pk], "search.RECAPDocument")
+        await sync_to_async(add_items_to_solr)([rd.pk], "search.RECAPDocument")
+
+    de_query = RECAPDocument.objects.filter(
+        docket_entry=de,
+        document_number=document_number,
+    )
 
-    mark_ia_upload_needed(de.docket, save_docket=True)
-    process_orphan_documents(
+    dupe_doc_ids = (
+        de_query.values("pacer_doc_id")
+        .filter()
+        .annotate(Count("id"))
+        .order_by()
+        .filter(id__count__gt=1)
+    )
+
+    if await dupe_doc_ids.aexists():
+        dupes = de_query.filter(
+            pacer_doc_id__in=[i["pacer_doc_id"] for i in dupe_doc_ids]
+        )
+        for dupe in dupes.aiterator():
+            for attachment in attachment_dicts:
+                attachment_number = attachment["attachment_number"]
+                pacer_doc_id = attachment["pacer_doc_id"]
+                if dupe.pacer_doc_id == pacer_doc_id:
+                    if dupe.attachment_number != attachment_number:
+                        await dupe.adelete()
```

Comment:
Could we centralize this with the other code, perhaps?

Reply:
I didn't see an obvious way to do that without introducing a bunch of extra queries within the loop. This way at least there's only a single duplicate check query in cases where there are no duplicates(which is of course the typical case).

Reply:
Gotcha. Let's pull this into its own method, then, and do a good job with the docstring so it's clear what this mess is all about.

Reply:
refactored into separate method